OrbitSsh/SftpFile.cpp

```diff
@@ -17,58 +17,44 @@ outcome::result<SftpFile> SftpFile::Open(Session* session, Sftp* sftp,
   if (result) {
     return outcome::success(SftpFile{result});
   } else {
-    return static_cast<SftpError>(
+    return static_cast<Error>(
         libssh2_session_last_errno(session->GetRawSessionPtr()));
   }
 }
 
 outcome::result<std::string> SftpFile::Read(int max_length_in_bytes) {
   std::string buffer(max_length_in_bytes, '\0');
   const auto result =
-      libssh2_sftp_read(file_ptr_, buffer.data(), buffer.size());
+      libssh2_sftp_read(file_ptr_.get(), buffer.data(), buffer.size());
 
   if (result >= 0) {
     buffer.resize(result);
     return buffer;
   } else {
-    return static_cast<SftpError>(result);
+    return static_cast<Error>(result);
   }
 }
 
 outcome::result<void> SftpFile::Close() {
-  const auto result = libssh2_sftp_close_handle(file_ptr_);
+  const auto result = libssh2_sftp_close_handle(file_ptr_.get());
 
   if (result == 0) {
-    file_ptr_ = nullptr;
+    file_ptr_.release();
```

Comment:
Just to be sure, this `release()` is correct? Aka it works as expected with the `std::unique_ptr<LIBSSH2_SFTP_HANDLE, decltype(&libssh2_sftp_close_handle)>` from SftpFile.h.

Reply:
Yes, release is correct here. We don't want unique_ptr to call the deleter, as this has been what we've just done.